### PR TITLE
bindinfo, planner: make evolve tasks consider runtime hints instead of ignoring them

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/15663, close https://github.com/pingcap/tidb/issues/15667 

Problem Summary:
1. evolve binding tasks ignore runtime hints while manually created bindings keep them
2. `explain format=hint` cannot extract runtime hints like `max_execution_time`

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: planner/core/hints.go [hunk 1]
```diff
@@ -165,8 +168,34 @@ func (p *BlockHintProcessor) getCurrentStmtHints(hints []*ast.TableOptimizerHint
 	return p.QbHints[currentOffset]
 }
 
+func restoreOptimizerHint(hint *ast.TableOptimizerHint) string {
```
Comment (lzmhhh123) [inline planner/core/hints.go]: It and `bindinfo.RestoreTableOptimizerHint` are the same.

Comment (qw4990) [reply planner/core/hints.go]: You are right. `restoreOptimizerHint` has been removed now, PTAL.

**Comment by qw4990** [comment]
@eurekaka PTAL

**Pull request merged**